### PR TITLE
iasm.d: rework to remove all storage allocation

### PR DESCRIPTION
**Pull request by WalterBright**
Replace new/delete with stack locals.

### CODE REVIEW

**Comment by WalterWaldron** [comment]
```std.concurrency``` test failure is [bug #17037](https://issues.dlang.org/show_bug.cgi?id=17037)

File: src/iasm.d [hunk 1]
```diff
@@ -3592,182 +3584,175 @@ int asm_getnum()
...
         asm_token();
-        o2 = asm_cond_exp();
+        OPND o2;
+        asm_cond_exp(o2);
         asm_chktok(TOKcolon,"colon");
-        o3 = asm_cond_exp();
-        o1 = (o1.disp) ? o2 : o3;
+        OPND o3;
+        asm_cond_exp(o3);
+        if (o1.disp)
```
Comment (andralex) [inline src/iasm.d]: use ternary op? `o1 = o1.disp ? o2 : o3;`

**Review by andralex — APPROVED** [review]

**Comment by andralex** [comment]
again seeing coverage issues

**Comment by WalterBright** [comment]
The coverage issues were there before, and are an issue for another PR.

**Comment by CyberShadow** [comment]
This pull request introduced a regression:
https://issues.dlang.org/show_bug.cgi?id=17523

**Pull request merged**